drivers/bluetooth/hci/hci_ifx_cyw208xx.c

```diff
@@ -185,6 +190,23 @@ static int cyw208xx_setup(const struct device *dev, const struct bt_hci_setup_pa
 
 	/* Waiting when BLE up after firmware launch */
 	cybt_platform_hci_wait_for_boot_fully_up(false);
+
+	/* Set public address */
+	buf = bt_hci_cmd_create(HCI_BRCM_SET_LOCAL_DEV_ADDR, BTM_SET_LOCAL_DEV_ADDR_LENGTH);
```

Comment:
You shouldn't unconditionally do this, but only if the host actually has a public address to provide to the driver. If it doesn't have a public address the value will be `BT_ADDR_ANY`, so all of the public address related code should be behind something like this:
```c
	if (!bt_addr_eq(&params->public_addr, BT_ADDR_ANY)) {
		...
	}
```

Reply:
Hi @jhedberg 

Why i do this without conditions?

cyw208xx's controller by default set some hardcoded static address, so it will same for all cyw208xx devices.

I added set public address without condition. in case if user do not call bt_id_create before calling bt_enable() to define address, the  params->public_addr will be Zeros, so we will overwrite current hardcoded 20:82….:93 address with Zero.. As result bt_id_init will fail bt_setup_public_id_addr (because zero address) and set set_random_address.

So used doesn't set bt_id_create before bt_enable, we will use random address setup by bt_id_init (colled from hci_init)

Let me know if you see any issue here?

Regards,
Nazar

Reply:
Thanks for the clarification. Please add a code comment then explaining that this is intentional and what happens when you get `BT_ADDR_ANY` from the host.

Reply:
updated here https://github.com/zephyrproject-rtos/zephyr/pull/76857/commits/d5d8b5bcc01839802b0eef5fccf94edafe51a0c5. 

Is it clear now?

Reply:
> updated here [d5d8b5b](https://github.com/zephyrproject-rtos/zephyr/commit/d5d8b5bcc01839802b0eef5fccf94edafe51a0c5).
> 
> Is it clear now?

Yes, looks much better now. Thanks!